src/pipelines/rust_app.rs

```diff
@@ -291,26 +292,59 @@ pub struct RustAppOutput {
     pub wasm_output: String,
 }
 
+pub fn pattern_evaluate(pattern: String, params: &HashMap<String, String>) -> String {
+    let mut result = pattern.clone();
+    for (k, v) in params.iter() {
+        let needle = "{".to_owned() + k.as_str() + "}";
+        if k.starts_with("@") {
+            if let Ok(contents) = std::fs::read_to_string(v.as_str()) {
+                result = str::replace(result.as_str(), &needle, contents.as_str());
+            }
+        } else {
+            result = str::replace(result.as_str(), &needle, v);
+        }
+    }
+    result
+}
+
 impl RustAppOutput {
     pub async fn finalize(self, dom: &mut Document) -> Result<()> {
         let (base, js, wasm, head, body) = (&self.cfg.public_url, &self.js_output, &self.wasm_output, "html head", "html body");
-
-        let preload = format!(
-            r#"
+        let (pattern_script, pattern_preload) = (&self.cfg.pattern_script, &self.cfg.pattern_preload);
+        let mut params: HashMap<String, String> = match &self.cfg.pattern_params {
+            Some(x) => x.clone(),
+            None => HashMap::new(),
+        };
```

Comment:
The gods of borrow checker doesn't actually like that piece of suggestion. I understand the intention, but can't seem to find the way to avoid cloning that map of parameters before its further modification. So probably will leave it how it was.